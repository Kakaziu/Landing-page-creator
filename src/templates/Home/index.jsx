import SectionBox from '../../components/SectionBox';

const Home = () => {
  return (
    <SectionBox
      background
      title="Criador de landing pages"
      paragraphs={[
        {
          text: 'Transforme sua visão em realidade! Crie uma landing page impressionante e conquiste a atenção do seu público. Experimente agora e dê vida à sua mensagem de forma envolvente. Comece a construir sua presença online hoje mesmo!',
        },
      ]}
      button={{
        text: 'Comece a criar',
        url: '#',
        size: 'big',
      }}
    />
  );
};

export default Home;
