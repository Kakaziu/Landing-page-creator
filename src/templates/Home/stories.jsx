import Home from '.';

export default {
  title: 'Home',
  component: Home,
};

export const Template = (args) => (
  <div>
    <Home {...args} />
  </div>
);
