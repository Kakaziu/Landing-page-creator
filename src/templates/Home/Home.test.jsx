import Home from '.';
import renderTheme from '../../styles/render-theme';

describe('render <Home />', () => {
  it('should render', () => {
    renderTheme(<Home />);
  });
});
