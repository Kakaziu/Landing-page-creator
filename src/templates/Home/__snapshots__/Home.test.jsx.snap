// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Home /> should render home 1`] = `
.c8 {
  color: #0A1128;
  text-decoration: none;
  font-size: 2.4rem;
  display: flex;
  align-items: center;
}

.c8 >img {
  max-width: 15rem;
  max-height: 8rem;
}

.c6 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c17 {
  color: #0A1128;
  text-transform: none;
  font-size: 2.4rem;
}

.c9 {
  color: #0A1128;
  display: flex;
  flex-flow: row wrap;
}

.c10 {
  color: #0A1128;
  font-size: 1.6rem;
  text-decoration: none;
  display: block;
  position: relative;
  padding: 1.6rem;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.5rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c3 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c1 {
  position: fixed;
  z-index: 5;
  padding: 10px 0px;
  width: 100%;
  background: #FFFFFF;
  transition: 0.3s;
  overflow-y: auto;
  border-bottom: 1px solid #DDDDDD;
}

.c1 .c2 {
  padding-top: 0;
  padding-bottom: 0;
}

.c0 {
  position: fixed;
  z-index: 6;
  top: 2rem;
  right: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  display: none;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c4 .c2 {
  padding-top: 0;
  padding-bottom: 0;
}

.c20 {
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 1px solid #DDDDDD;
  font-size: 1.6rem;
  padding: 1.6rem;
}

.c20 a {
  color: #0A1128;
  text-decoration: none;
  font-size: 1.6rem;
}

.c20 .c2 {
  padding: 1.6rem 0;
}

.c21 {
  position: fixed;
  bottom: 4rem;
  right: 4rem;
  width: 40px;
  height: 40px;
  border-radius: 20px;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(10, 18, 40, 0.5);
  transition: all 0.3s ease-in-out;
  cursor: pointer;
}

.c21:hover {
  background-color: #0A1128;
}

.c11 {
  background-color: #0A1128;
  color: #FFFFFF;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c18 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c16 .c5 {
  margin-bottom: 1.6rem;
}

.c13 .c5 {
  margin-bottom: 1.6rem;
}

.c15 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 4.0rem;
  margin-top: 5.6rem;
  width: 100%;
}

.c12 {
  margin-top: 8rem;
}

.c12 .c5 {
  margin-bottom: 1.6rem;
  color: #FFFFFF;
}

.c19 {
  margin-top: 8rem;
}

.c19 .c5 {
  margin-bottom: 1.6rem;
  color: #0A1128;
}

.c14 {
  font-size: 1.6rem;
}

@media (max-width:768px) {
  .c9 {
    flex-flow: column nowrap;
    align-items: center;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c1 {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    visibility: hidden;
    opacity: 0;
  }

  .c1 .c7 {
    margin-bottom: 3rem;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

@media (max-width:768px) {
  .c4 {
    display: block;
    height: 42rem;
    overflow-y: auto;
    padding: 3.2rem;
    width: 100%;
  }

  .c4::-webkit-scrollbar {
    width: 5px;
  }

  .c4::-webkit-scrollbar-thumb {
    background-color: #0A1128;
    border-radius: 50px;
  }
}

@media (max-width:768px) {
  .c15 {
    grid-template-columns: 1fr;
  }
}

<div>
  <button
    aria-label="Open/Close Menu"
    class="c0"
  >
    <svg
      aria-label="Open Menu"
      color="#dc143c"
      fill="currentColor"
      height="35"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(220, 20, 60);"
      viewBox="0 0 24 24"
      width="35"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c1"
    visible="0"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <h1
          class="c5 c6"
          isdark="1"
          uppercase="1"
        >
          <a
            class="c7 c8"
            href="#"
          >
            <span>
              Logo
            </span>
          </a>
        </h1>
        <nav
          class="c9"
        >
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c10"
            href="#target2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c10"
            href="#target3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="c10"
            href="#target4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c10"
            href="#target5"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="c10"
            href="#target6"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="c10"
            href="#target7"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="c10"
            href="#target8"
            target="_self"
          >
            Link 8
          </a>
          <a
            class="c10"
            href="#target9"
            target="_self"
          >
            Link 9
          </a>
          <a
            class="c10"
            href="#target10"
            target="_self"
          >
            Link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
  <div
    background="1"
    class="c11"
  >
    <div
      class="c2 c3"
    >
      <div
        background="1"
        class="c12"
        id=""
      >
        <div
          class="c13"
        >
          <h1
            class="c5 c6"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c14"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 1
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 2
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 3
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    background="0"
    class="c18"
  >
    <div
      class="c2 c3"
    >
      <div
        background="0"
        class="c19"
        id=""
      >
        <div
          class="c13"
        >
          <h1
            class="c5 c6"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c14"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 1
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 2
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 3
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    background="1"
    class="c11"
  >
    <div
      class="c2 c3"
    >
      <div
        background="1"
        class="c12"
        id=""
      >
        <div
          class="c13"
        >
          <h1
            class="c5 c6"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c14"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 1
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 2
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 3
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    background="0"
    class="c18"
  >
    <div
      class="c2 c3"
    >
      <div
        background="0"
        class="c19"
        id=""
      >
        <div
          class="c13"
        >
          <h1
            class="c5 c6"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c14"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 1
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 2
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 3
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    background="1"
    class="c11"
  >
    <div
      class="c2 c3"
    >
      <div
        background="1"
        class="c12"
        id=""
      >
        <div
          class="c13"
        >
          <h1
            class="c5 c6"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c14"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 1
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 2
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 3
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    background="0"
    class="c18"
  >
    <div
      class="c2 c3"
    >
      <div
        background="0"
        class="c19"
        id=""
      >
        <div
          class="c13"
        >
          <h1
            class="c5 c6"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c14"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 1
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 2
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
              isdark="1"
              uppercase="0"
            >
              Teste 3
            </h1>
            <p
              class="c14"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="c20"
  >
    <p>
      Teste de footer
    </p>
  </footer>
  <a
    aria-label="Go to top"
    class="c21"
    href="#"
    title="Go to top"
  >
    <svg
      color="white"
      fill="currentColor"
      height="35"
      stroke="currentColor"
      stroke-width="0"
      style="color: white;"
      viewBox="0 0 512 512"
      width="35"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128 320l128-128 128 128z"
      />
    </svg>
  </a>
</div>
`;
